.github/workflows/ci.yaml

```diff
@@ -0,0 +1,240 @@
+          build/target_graph.dot
+          build/qbittorrent.exe
+          build/qbittorrent.pdb
+          dist/windows/qt.conf
+
+  ci_macos:
+    name: macOS + vcpkg
+
+    strategy:
+      matrix:
+        qbt_gui: ["GUI=ON", "GUI=OFF"]
+      fail-fast: false
+
+    runs-on: macos-10.15
+
+    defaults:
+      run:
+        shell: pwsh
+
+    steps:
+    - name: checkout repository
+      uses: actions/checkout@v2.3.2
+
+    # - ninja is needed for building qBittorrent (because it's preferrable, not a hard requirement)
+    # - pkg-config is needed for some vcpkg installations
+    - name: install additional required packages with homebrew
+      shell: bash
+      run: |
+        brew install ninja pkg-config
+
+    - name: setup vcpkg (cached, if possible)
+      uses: lukka/run-vcpkg@v3.3
+      with:
+        vcpkgDirectory: ${{ env.VCPKG_DEST_MACOS }}
+        vcpkgGitCommitId: ${{ env.VCPKG_COMMIT }}
+        setupOnly: true
+
+    - name: configure vcpkg triplet overlay for release builds only
+      run: |
+        New-Item -Path ${{ github.workspace }} -Name "triplets_overlay" -ItemType Directory
+        Copy-Item ${{ env.RUNVCPKG_VCPKG_ROOT }}/triplets/x64-osx.cmake `
+          ${{ github.workspace }}/triplets_overlay/x64-osx-release.cmake
+        Add-Content ${{ github.workspace }}/triplets_overlay/x64-osx-release.cmake `
+          -Value "set(VCPKG_BUILD_TYPE release)"
+
+    - name: force AppleClang to compile libtorrent with C++14
+      run: |
+        (Get-Content -path ${{ env.RUNVCPKG_VCPKG_ROOT }}/ports/libtorrent/portfile.cmake).Replace( `
+          '${FEATURE_OPTIONS}', '${FEATURE_OPTIONS} -DCMAKE_CXX_STANDARD=14') `
+          | Set-Content -Path ${{ env.RUNVCPKG_VCPKG_ROOT }}/ports/libtorrent/portfile.cmake
+
+    - name: install dependencies via vcpkg
+      run: |
+        $packages = `
+          "boost-circular-buffer:x64-osx-release",
+          "libtorrent:x64-osx-release",
+          "qt5-base:x64-osx-release",
+          "qt5-svg:x64-osx-release",
+          "qt5-tools:x64-osx-release",
+          "qt5-macextras:x64-osx-release"
+        foreach($package in $packages)
+        {
+          ${{ env.RUNVCPKG_VCPKG_ROOT }}/vcpkg install $package `
+            --overlay-triplets=${{ github.workspace }}/triplets_overlay `
+            --clean-after-build
+        }
+
+    - name: build qBittorrent
+      shell: bash
+      run: |
+        cmake -B build -G "Ninja" -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -DCMAKE_TOOLCHAIN_FILE=${{ env.VCPKG_DEST_MACOS }}/scripts/buildsystems/vcpkg.cmake \
+          -DVCPKG_TARGET_TRIPLET=x64-osx-release \
+          -D${{ matrix.qbt_gui }} \
+          -DVERBOSE_CONFIGURE=ON \
+          --graphviz=build/target_graph.dot
+        cmake --build build
+
+    - name: upload artifact as zip
+      uses: actions/upload-artifact@v2.1.3
+      with:
+        name: qBittorrent-CI-macOS_x64-static-release_${{ matrix.qbt_gui }}
```

Comment:
@Chocobo1 

@Kolcha had tested the macOS builds during the development of the CMake PR, and found that CMake correctly generated the app bundle and that the executable ran.
The relevant code hasn't been modified since, and the CI builds macOS builds the same way as one would in their machine.

Plus, as mentioned in https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491119532, there are other things that are included in the artifact archive that are useful, not just the executable; and the executable generated by this workflow is not meant for general use, just for troubleshooting and smoke testing purposes (e.g. does it run?).

Reply:
OK.